internal/servers/destinations.go

```diff
@@ -89,7 +89,13 @@ func (s *DestinationServer) Write(msg pb.Destination_WriteServer) error {
 			wg.Wait()
 			return status.Errorf(codes.InvalidArgument, "failed to unmarshal spec: %v", err)
 		}
-		resources <- resource
+		select {
+		case resources <- resource:
+		case <-msg.Context().Done():
+			close(resources)
```

Comment:
I would add these two lines as a defer statement and remove multiple copies of it here and above.

Reply:
That won't work because when `err == io.EOF` I want wait for `write` so we can finish writing and send the writeSummary